substrate/frame/bounties/src/benchmarking.rs

```diff
@@ -17,87 +17,163 @@
+use crate::{tests::utils::*, Pallet as Bounties};
 
 use alloc::{vec, vec::Vec};
 use frame_benchmarking::v1::{
 	account, benchmarks_instance_pallet, whitelisted_caller, BenchmarkError,
 };
+use frame_support::traits::Currency;
 use frame_system::{pallet_prelude::BlockNumberFor as SystemBlockNumberFor, RawOrigin};
-use sp_runtime::traits::{BlockNumberProvider, Bounded};
-
-use crate::Pallet as Bounties;
 use pallet_treasury::Pallet as Treasury;
+use sp_core::crypto::FromEntropy;
+use sp_runtime::traits::BlockNumberProvider;
 
-const SEED: u32 = 0;
-
-fn set_block_number<T: Config<I>, I: 'static>(n: BlockNumberFor<T, I>) {
-	<T as pallet_treasury::Config<I>>::BlockNumberProvider::set_block_number(n);
+/// Trait describing factory functions for dispatchables' parameters.
+pub trait ArgumentsFactory<AssetKind> {
+	/// Factory function for an asset kind.
+	fn create_asset_kind(seed: u32) -> AssetKind;
 }
 
-fn minimum_balance<T: Config<I>, I: 'static>() -> BalanceOf<T, I> {
-	let minimum_balance = T::Currency::minimum_balance();
-
-	if minimum_balance.is_zero() {
-		1u32.into()
-	} else {
-		minimum_balance
+/// Implementation that expects the parameters implement the [`FromEntropy`] trait.
+impl<AssetKind> ArgumentsFactory<AssetKind> for ()
+where
+	AssetKind: FromEntropy,
+{
+	fn create_asset_kind(seed: u32) -> AssetKind {
+		AssetKind::from_entropy(&mut seed.encode().as_slice()).unwrap()
 	}
 }
 
-// Create bounties that are approved for use in `on_initialize`.
-fn create_approved_bounties<T: Config<I>, I: 'static>(n: u32) -> Result<(), BenchmarkError> {
-	for i in 0..n {
-		let (caller, _curator, _fee, value, reason) =
-			setup_bounty::<T, I>(i, T::MaximumReasonLength::get());
-		Bounties::<T, I>::propose_bounty(RawOrigin::Signed(caller).into(), value, reason)?;
-		let bounty_id = BountyCount::<T, I>::get() - 1;
-		let approve_origin =
-			T::SpendOrigin::try_successful_origin().map_err(|_| BenchmarkError::Weightless)?;
-		Bounties::<T, I>::approve_bounty(approve_origin, bounty_id)?;
-	}
-	ensure!(BountyApprovals::<T, I>::get().len() == n as usize, "Not all bounty approved");
-	Ok(())
+const SEED: u32 = 0;
+
+fn set_block_number<T: Config<I>, I: 'static>(n: BlockNumberFor<T, I>) {
+	<T as pallet_treasury::Config<I>>::BlockNumberProvider::set_block_number(n);
 }
 
 // Create the pre-requisite information needed to create a treasury `propose_bounty`.
 fn setup_bounty<T: Config<I>, I: 'static>(
-	u: u32,
-	d: u32,
-) -> (T::AccountId, T::AccountId, BalanceOf<T, I>, BalanceOf<T, I>, Vec<u8>) {
-	let caller = account("caller", u, SEED);
-	let value: BalanceOf<T, I> = T::BountyValueMinimum::get().saturating_mul(100u32.into());
-	let fee = value / 2u32.into();
+	user: u32,
+	description: u32,
+) -> (
+	T::AccountId,
+	T::AccountId,
+	T::AssetKind,
+	BountyBalanceOf<T, I>,
+	BountyBalanceOf<T, I>,
+	T::Beneficiary,
+	Vec<u8>,
+) {
+	let caller = account("caller", user, SEED);
+	// Tiago: check with Muharem if we need coupling with pallet-assets
```

Comment:
no, we should not, otherwise you can only use thus pallet if you use pallet-assets